File: ECommerce/src/main/java/com/satouhikaru/service/CustomerService.java
package com.satouhikaru.service;

import com.satouhikaru.entity.Customer;

import java.util.List;

/**
 * @author Pham Nguyen Ha Quang
 * @since  17/11/2018
 */
public interface CustomerService {

	/**
	 * Count list of customers quantity
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  29/12/2018
	 *
	 * @return long
	 */
	long countAll();

	/**
	 * Count list of customers quantity by key search
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  29/12/2018
	 *
	 * @param keySearch
	 * @return long
	 */
	long countByKeySearch(String keySearch);

	/**
	 * Get list of customers per page
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  15/12/2018
	 *
	 * @param startRow
	 * @param maxRows
	 * @return List
	 */
	List<Customer> getPerPage(long startRow, long maxRows);

	/**
	 * Get list of customers by name
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  15/12/2018
	 *
	 * @param keySearch
	 * @param startRow
	 * @param maxRows
	 * @return List
	 */
	List<Customer> getByKeySearch(String keySearch, long startRow, long maxRows);

	/**
	 * Get customer by ID
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  15/12/2018
	 *
	 * @param id customer ID
	 * @return Customer
	 */
	Customer getById(long id);

	/**
	 * Get customer by user name
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  15/12/2018
	 *
	 * @param userName customer's user name
	 * @return Customer
	 */
	Customer getByUserName(String userName);

	/**
	 * Add new customer
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  15/12/2018
	 *
	 * @param customer a new customer
	 */
	void add(Customer customer);

	/**
	 * Update customer
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  15/12/2018
	 *
	 * @param customer customer need updating
	 */
	void update(Customer customer);
	
}

File: Angular/src/app/module/account/components/customer/service/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { User } from '../../../model/user.class';
import { Proxy } from 'src/app/common/proxy.class';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  url = this.proxyUrl.baseUrl.concat('/customers');

  constructor(private http: HttpClient, private proxyUrl: Proxy) { }

  /**
   * Count list of customer quantity
   * 
   * @since 29/12/2018
   */
  public countAll() {
    return this.http.get<number>(this.url + '/countAll');
  }

  /**
   * Count customers quantity
   * 
   * @since 02/01/2018
   */
  public countByKeySearch(keySearch: string) {
    return this.http.get<number>(this.url + '/count?keySearch=' + keySearch);
  }

  /**
   * Get list of customers per page
   *
   * @since 02/01/2018
   * @param page
   */
  public getPerPage(page: number) {
    return this.http.get<User[]>(this.url + '/page/' + page);
  }

  /**
   * Get list of customers by key search
   *
   * @since 02/01/2018
   * @param key search
   * @param page
   */
  public getByKeySearch(keySearch: string, page: number) {
    return this.http.get<User[]>(this.url + '/search/' + page + '?keySearch=' + keySearch);
  }

  /**
   * Get customer by ID
   *
   * @since 02/01/2018
   * @param id
   */
  public getById(id: number) {
    return this.http.get<User>(this.url + '/id/' + id);
  }

  /**
   * Get customer by user name
   *
   * @since 02/01/2018
   * @param userName
   */
  public getByUserName(userName: string) {
    return this.http.get<User>(this.url + '/userName/' + userName);
  }

  /**
   * Add new customer
   * 
   * @since 02/01/2018
   * @param customer new customer
   */
  public add(customer: User) {
    return this.http.post(this.url, customer);
  }

  /**
   * Update customer
   * 
   * @since 02/01/2018
   * @param customer customer need updating
   */
  public update(customer: User) {
    return this.http.put(this.url + '/' + customer.id, customer);
  }

  /**
   * Delete customer
   * 
   * @since 02/01/2018
   * @param customerName customer name
   */
  public delete(id: number) {
    return this.http.delete(this.url + '/' + id);
  }

}